ci/Jenkinsfile

```diff
@@ -0,0 +1,100 @@
+@Library('jenkins_shared_lib') _
+
+pipeline {
+  agent any
+  environment {
+    BUILD_TYPE = sh(returnStdout: true, script: 'rapids-build-type')
+  }
+  options {
+    buildDiscarder(logRotator(numToKeepStr: '10'))
+    ansiColor('xterm')
+    disableConcurrentBuilds(abortPrevious: true)
+  }
+  stages {
+    stage("Style Checks") {
+      steps {
+        sh "env"
+      }
+    }
+    stage("C++ Matrix Build") {
+      environment {
+        PARALLEL_LEVEL = '4'
+      }
+      options {
+        timeout(time: 30, unit: 'MINUTES', activity: true)
+      }
+      steps {
+        script {
+          // generate the CUDA stages across different configuration options
+          parallel generateStage("cuda_build", {
+            sh "./ci/build_cpp.sh"
+          })
+        }
+      }
+    }
+
+    stage("Python Matrix Build") {
+      options {
+        timeout(time: 30, unit: 'MINUTES', activity: true)
+      }
+      steps {
+        script {
+          // generate the python stages across different configuration options
+          parallel generateStage("python_build", {
+            sh "./ci/build_py.sh"
+          })
+        }
+      }
+    }
+
+    stage("PR/Branch Test Stage") {
+      when { not { triggeredBy 'TimerTrigger' } }
```

Comment:
Since we're no longer going to run tests on the branch builds, we can rename this stage to just `PR Test Stage`.

Also, let's switch all of these `when` statements to use the `BUILD_TYPE` variable.

As a reminder, here's what should happen for the different build types:

- PR: test, but don't upload
- Branch: don't test, but upload
- Nightly: test and upload